server/core/region_store.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 TiKV Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+)
+
```

Comment:
Shall we define something like below then implement all the following methods as RegionStore's?
`
type RegionStore struct {
        RegionTree
}
`